File: solutions/two-sum.cpp
// Author: Joshua Obogbaimhe
// Problem https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> numMap; 
        
        for(int i = 0; i < nums.size(); i++) {
            int val = target - nums[i]; 
            int numCount = numMap.count(val); 
            
            if(numCount) {
                return {i, numMap[val]}; 
            }
            
            numMap[nums[i]] = i; 
        }
        
        return {}; 
    }
};

File: solutions/javascript/two-sum.js
// Author: Carlos Duque
//Problem: https://leetcode.com/problems/two-sum/

var twoSum = function (nums, target) {
  if (nums.length === 2) {
    return [0, 1];
  }

  var hashTable = {};

  for (var i = 0; i < nums.length; i++) {
    var val = target - nums[i];

    if (val in hashTable) {
      return [i, hashTable[val]];
    }

    hashTable[nums[i]] = i;
  }
};
